<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>816</width>
    <height>626</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Hệ thống phân loại khối lượng</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_Left">
      <item>
       <widget class="QGroupBox" name="groupBoxSerialConfig">
        <property name="title">
         <string>Cấu hình kết nối UART</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="labelPort">
           <property name="text">
            <string>COM Port:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="comboBoxPort"/>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="pushButtonRefreshPorts">
           <property name="text">
            <string>Refresh</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="labelBaudRate">
           <property name="text">
            <string>Baud Rate:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="comboBoxBaudRate"/>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="labelDataBits">
           <property name="text">
            <string>Data Bits:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="comboBoxDataBits"/>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="labelStopBits">
           <property name="text">
            <string>Stop Bits:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QComboBox" name="comboBoxStopBits"/>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="labelParity">
           <property name="text">
            <string>Parity:</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QComboBox" name="comboBoxParity"/>
         </item>
         <item row="5" column="0" colspan="3">
          <layout class="QHBoxLayout" name="horizontalLayoutConnect">
           <item>
            <widget class="QPushButton" name="pushButtonConnect">
             <property name="text">
              <string>Kết nối</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButtonDisconnect">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Ngắt kết nối</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBoxSystemConfig">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="title">
         <string>Cấu hình hệ thống</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="labelThreshold1">
           <property name="text">
            <string>Threshold 1:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="spinBoxThreshold1">
           <property name="maximum">
            <number>9999</number>
           </property>
           <property name="value">
            <number>100</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="labelThreshold2">
           <property name="text">
            <string>Threshold 2:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="spinBoxThreshold2">
           <property name="maximum">
            <number>9999</number>
           </property>
           <property name="value">
            <number>500</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="labelLightPreset">
           <property name="text">
            <string>Light Preset:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="spinBoxLightPreset">
           <property name="maximum">
            <number>9999</number>
           </property>
           <property name="value">
            <number>1000</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="labelMediumPreset">
           <property name="text">
            <string>Medium Preset:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QSpinBox" name="spinBoxMediumPreset">
           <property name="maximum">
            <number>9999</number>
           </property>
           <property name="value">
            <number>2000</number>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="labelHeavyPreset">
           <property name="text">
            <string>Heavy Preset:</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QSpinBox" name="spinBoxHeavyPreset">
           <property name="maximum">
            <number>9999</number>
           </property>
           <property name="value">
            <number>3000</number>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="labelPower">
           <property name="text">
            <string>Power:</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <layout class="QHBoxLayout" name="horizontalLayoutPower">
           <item>
            <widget class="QPushButton" name="pushButtonPowerOn">
             <property name="text">
              <string>Bật nguồn</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButtonPowerOff">
             <property name="text">
              <string>Tắt nguồn</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="6" column="0" colspan="2">
          <widget class="QPushButton" name="pushButtonSendConfig">
           <property name="text">
            <string>Gửi cấu hình</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_Right">
      <item>
       <widget class="QGroupBox" name="groupBoxSystemStatus">
        <property name="title">
         <string>Trạng thái hệ thống</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="QLabel" name="labelStatusThreshold1">
           <property name="text">
            <string>Threshold 1:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLCDNumber" name="lcdNumberThreshold1">
           <property name="digitCount">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="labelStatusThreshold2">
           <property name="text">
            <string>Threshold 2:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QLCDNumber" name="lcdNumberThreshold2">
           <property name="digitCount">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="labelTotalCount">
           <property name="text">
            <string>Tổng số lượng:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLCDNumber" name="lcdNumberTotalCount">
           <property name="digitCount">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="labelLastWeight">
           <property name="text">
            <string>Khối lượng gần nhất:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QLCDNumber" name="lcdNumberLastWeight">
           <property name="digitCount">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="labelSystemStatusTitle">
           <property name="text">
            <string>Trạng thái:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="labelSystemStatus">
           <property name="text">
            <string>Stopped</string>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QLabel" name="labelFullStatusTitle">
           <property name="text">
            <string>Thùng đầy:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <widget class="QLabel" name="labelFullStatus">
           <property name="text">
            <string>NO</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBoxBins">
        <property name="title">
         <string>Tình trạng các thùng</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0" colspan="2">
          <widget class="QLabel" name="labelLightBin">
           <property name="text">
            <string>Light Bin</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="2" colspan="2">
          <widget class="QLabel" name="labelMediumBin">
           <property name="text">
            <string>Medium Bin</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="4" colspan="2">
          <widget class="QLabel" name="labelHeavyBin">
           <property name="text">
            <string>Heavy Bin</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="labelLightPresetStatus">
           <property name="text">
            <string>Preset:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLCDNumber" name="lcdNumberLightPreset">
           <property name="digitCount">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="labelMediumPresetStatus">
           <property name="text">
            <string>Preset:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QLCDNumber" name="lcdNumberMediumPreset">
           <property name="digitCount">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="1" column="4">
          <widget class="QLabel" name="labelHeavyPresetStatus">
           <property name="text">
            <string>Preset:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="5">
          <widget class="QLCDNumber" name="lcdNumberHeavyPreset">
           <property name="digitCount">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="labelLightCurrentStatus">
           <property name="text">
            <string>Current:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLCDNumber" name="lcdNumberLightCurrent">
           <property name="digitCount">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QLabel" name="labelMediumCurrentStatus">
           <property name="text">
            <string>Current:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <widget class="QLCDNumber" name="lcdNumberMediumCurrent">
           <property name="digitCount">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="2" column="4">
          <widget class="QLabel" name="labelHeavyCurrentStatus">
           <property name="text">
            <string>Current:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="5">
          <widget class="QLCDNumber" name="lcdNumberHeavyCurrent">
           <property name="digitCount">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0" colspan="2">
          <widget class="QProgressBar" name="progressBarLight">
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item row="3" column="2" colspan="2">
          <widget class="QProgressBar" name="progressBarMedium">
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item row="3" column="4" colspan="2">
          <widget class="QProgressBar" name="progressBarHeavy">
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBoxLog">
        <property name="title">
         <string>Log UART</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QTextEdit" name="textEditLog">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButtonClearLog">
           <property name="text">
            <string>Xóa log</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>816</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
